app/assets/javascripts/discourse/models/user.js.es6

```diff
@@ -619,6 +619,15 @@ const User = RestModel.extend({
     return ajax(`${userPath(this.get("username"))}/notification_level.json`, {
       type: "PUT",
       data: { notification_level: level, expiring_at: expiringAt }
+    }).then(() => {
+      const currentUser = Discourse.User.current();
```

Comment:
Hmm, isn't this.currentUser available? We try to avoid using globals as possible

Reply:
1. There is no `currentUser` property (we can access the `this` of the object which is the current model's context)
2. The `this` in the context of the User model doesn't actually refer to the correct version of the `currentUser`, the correct version of the currentUser is stored in the preloaded store and fetched via `Discourse.User.current()` ... in this same `user.js.es6` model file, you will find other parts using the `Discourse.User.current()` because that's the real version
3. You will also notice within the `user.js.es6` file, sometimes we do double updates to the `Discourse.User.current()` and the this (the model user's context)

So I think this is fine.